akka-stream/src/main/scala/akka/stream/impl/FanoutProcessor.scala

```diff
@@ -126,7 +126,7 @@ import org.reactivestreams.Subscriber
       log.debug("fail due to: {}", e.getMessage)
     primaryInputs.cancel()
     primaryOutputs.error(e)
-    // Stopping will happen after flush
+    context.stop(self)
   }
```

Comment:
now this is identical to super.fail in ActorProcessorImpl

Do we know why was it overridden in the first place?

Reply:
why doesn't the the `def afterShutdown(): Unit = afterFlush()` work?

Reply:
Perhaps @johanandren missed the comments. :)

Reply:
`afterFlush()` is a local method only called from `FanoutOutputs.afterShutdown` which in turn is only called from `FanoutOutputs.shutdown` and that is only invoked when downstream completes, or when the last subscriber unregisters, in the scenario of no registered subscribers-yet none of those two happens. 

I first thought that it was only for the no-registered downstreams scenario, but the added tests shows that it does in fact never work as implied by the comment that was here, even with subscribers. I'm not sure how is supposed to work - debugging I can see that no further callbacks whatsoever happens after that `failure`. Explicitly completing seems to work and doesn't not break the tck or any other tests, so I'm hoping this is fine.